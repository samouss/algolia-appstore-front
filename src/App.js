import React from 'react';
import { Provider } from 'components/Algolia';
import Layout from 'components/Layout';
import Filters from 'components/Filters';
import AdvancedFilters from 'components/AdvancedFilters';
import SearchFilter from 'containers/SearchFilter';
import CategoryFilter from 'containers/CategoryFilter';
import RatingFilter from 'containers/RatingFilter';
import RatingSort from 'containers/RatingSort';
import ClearFilter from 'containers/ClearFilter';
import InfiniteAppList from 'containers/InfiniteAppList';
import configureAlgolia from './configureAlgolia';

const algolia = configureAlgolia();

const App = () => (
  <Provider
    {...algolia}
  >
    <Layout>
      <Filters>
        <SearchFilter />

        <AdvancedFilters>
          <CategoryFilter />
          <RatingFilter />
          <RatingSort />
          <ClearFilter />
        </AdvancedFilters>
      </Filters>

      <InfiniteAppList />
    </Layout>
  </Provider>
);

export default App;
