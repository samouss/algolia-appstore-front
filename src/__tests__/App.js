import React from 'react';
import { shallow } from 'enzyme';
import App from '../App';

describe('<App>', () => {
  it('expect to render', () => {
    const component = shallow(
      <App />,
    );

    expect(component).toMatchSnapshot();
  });
});
