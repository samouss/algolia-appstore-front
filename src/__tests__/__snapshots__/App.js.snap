// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App> expect to render 1`] = `
<Provider
  apiKey="f8dac63a6dc38d8bd5893dad8969d819"
  appId="RTMJH2ID8J"
  indexName="apps_rating_desc"
  options={
    Object {
      "disjunctiveFacets": Array [
        "category",
        "rating",
      ],
    }
  }
>
  <Layout>
    <Filters>
      <connect(withQuery(SearchFilter)) />
      <AdvancedFilters>
        <connect(withFacet(connect(withDisjunctiveFacetRefinement(CategoryList)))) />
        <connect(withFacet(connect(withNumericRefinement(RatingList)))) />
        <connect(withIndex(RatingSort)) />
        <connect(withRefinements(connect(withClearRefinements(ClearList)))) />
      </AdvancedFilters>
    </Filters>
    <connect(withInfiniteHits(AppList)) />
  </Layout>
</Provider>
`;
